lib/app/features/feed/create_post/views/pages/create_post_modal/components/video_preview_cover.dart

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/components/screen_offset/screen_side_offset.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/core/providers/video_player_provider.dart';
+import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/components/video_preview_duration.dart';
+import 'package:ion/app/features/feed/create_post/views/pages/create_post_modal/components/video_preview_edit_cover.dart';
+import 'package:ion/app/features/gallery/providers/gallery_provider.dart';
+import 'package:ion/generated/assets.gen.dart';
+import 'package:video_player/video_player.dart';
+
+class VideoPreviewCover extends HookConsumerWidget {
+  const VideoPreviewCover({
+    required this.videoPath,
+    super.key,
+  });
+
+  final String videoPath;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final assetFilePathAsync = ref.watch(assetFilePathProvider(videoPath));
+    final filePath = assetFilePathAsync.valueOrNull;
+
+    if (filePath == null) {
+      return const _VideoPlaceholder();
+    }
+
+    final videoController = ref.watch(
+      videoControllerProvider(
+        filePath,
+        looping: true,
+      ),
+    );
+
+    if (!videoController.value.isInitialized) {
+      return const _VideoPlaceholder();
+    }
+
+    useEffect(
```

Comment:
Looks strange. You create the `videoController` using Riverpod, but dispose it with Hooks. It's better to move disposing to Riverpod provider.